src/PrestaShopBundle/Controller/Admin/ProductController.php

```diff
@@ -620,6 +608,45 @@ public function formAction($id, Request $request)
         );
     }
 
+    /**
+     * Builds the product form
+     *
+     * @param Product           $product
+     * @param AdminModelAdapter $modelMapper
+     * @return FormInterface
+     */
+    private function createProductForm (Product $product, AdminModelAdapter $modelMapper)
+    {
+        $formBuilder = $this->createFormBuilder(
+            $modelMapper->getFormData($product),
+            array('allow_extra_fields' => true, 'csrf_protection' => false)
+        )
+            ->add('id_product', 'Symfony\Component\Form\Extension\Core\Type\HiddenType')
+            ->add('step1', 'PrestaShopBundle\Form\Admin\Product\ProductInformation')
+            ->add('step2', 'PrestaShopBundle\Form\Admin\Product\ProductPrice')
+            ->add('step3', 'PrestaShopBundle\Form\Admin\Product\ProductQuantity')
+            ->add('step4', 'PrestaShopBundle\Form\Admin\Product\ProductShipping')
+            ->add('step5', 'PrestaShopBundle\Form\Admin\Product\ProductSeo', array(
+                'mapping_type' => $product->getRedirectType(),
+            ))
+            ->add('step6', 'PrestaShopBundle\Form\Admin\Product\ProductOptions');
+
+        // Prepare combination form (fake but just to validate the form)
+        $combinations = $product->getAttributesResume(
+            $this->get('prestashop.adapter.legacy.context')->getContext()->language->id
+        );
+        if (is_array($combinations)) {
```

Comment:
```diff
- is_array($combinations)
+ is_array($combinations) && count($combinations) > 0
```

Reply:
If we got an error, I'd have approved your answer.
But is it really important? In the case `count() === 0`, the foreach() won't loop but exit immediately.

Reply:
We don't write the code only for the PHP engine, it's only me but It's easier to read and understand what is done here, the best would be to check if $combinations is always an array (I think it is) and only keep the check on the count, wdyt?